_posts/2020-07-08-pull-requests-require-context.md

```diff
@@ -0,0 +1,148 @@
+changes.
+
+[Here's a PR](https://github.com/artsy/force/pull/5817) from [Ashley](https://github.com/ashleyjelks) that includes
+animated gifs of the changes she's made. The effects of the changes might not be obvious by looking only at the
+code, but seeing them in action makes it clear.
+
+### Document the unseen
+
+Sometimes a PR's changes have effects outside of the UI. There are still ways to give reviewers proof that the
+changes have the desired effects.
+
+[Here's a PR](https://github.com/artsy/eigen/pull/3206) from [Yuki](https://github.com/yuki24) that not only
+demonstrates what's happening in the UI, but also assures me that the back-end data is getting updated properly
+through a Rails console.
+
+[Christina](https://github.com/xtina-starr) authored [this PR](https://github.com/artsy/reaction/pull/3441) which
+shows the UI changes in addition to some output from her browser console, demonstrating that analytics tracking
+calls are firing correctly.
+
+### Share your progress
+
+One mistake many engineers make with non-trivial pull requests is to wait to open them until they're "done". If
+there are changes you'd like to get people's eyes on quickly, open a PR before the work is done and mark it as
+in-progress with`WIP` in the title. This helps avoid rework by starting early discussions about your approach.
+
+Let reviewers know in the body that your work isn't complete. As you continue your work, use a `TODO:` list in the
+body to illustrate your progress, as in [this PR](https://github.com/artsy/palette/pull/464) from
+[Sepand](https://github.com/sepans).
+
+## Spread knowledge
+
+Pull requests should not be one-sided - they aren't just about collecting feedback from the reviewer. They're also
+an opportunity to spread knowledge from the author.
+
+### Share your learnings
+
+Maybe you learned some things about the system you're working with, or you learned a new feature of the language.
+Share this new information with your team. [Roop](https://github.com/anandaroop) shares some findings about
+disabled tests [in this PR](https://github.com/artsy/metaphysics/pull/2130).
+
+We introduced a new state management library shortly before [I](https://github.com/pepopowitz) opened
+[this PR](https://github.com/artsy/eigen/pull/3526), and I had to do some reading about how to add types to
+something. I [shared my learnings with the team](https://github.com/artsy/eigen/pull/3526#discussion_r451161406).
+
+### Share development tips
+
+Did you learn a new technique while building this feature? Share it with your team!
+
+In [this PR](https://github.com/artsy/reaction/pull/3279#discussion_r395461329),
+[Chris](https://github.com/damassi) shared with us his technique for grabbing fixture data from his locally running
+environment.
+
+---
+
+Providing full context in a pull request will improve your asynchronous code reviews significantly. There are many
```

Comment:
Slightly off-topic: I love more context in a pull request, and this reminds me of the opposite problem: too many (potentially unrelated) problems to address in a single PR, and dividing them into focused/digestible PRs makes reviewing easier. This may be relevant to the "Define the problem and solution" section above.